src/main/java/com/puppycrawl/tools/checkstyle/checks/ArrayTypeStyleCheck.java

```diff
@@ -147,8 +147,7 @@ public void visitToken(DetailAST ast) {
             // force all methods to be Java style (see note in top Javadoc)
             final boolean isMethodViolation = isMethod && !isJavaStyle;
             final boolean isVariableViolation = !isMethod
-                    && isJavaStyle != javaStyle
-                    && typeAST.getType() != TokenTypes.TYPE_ARGUMENT;
```

Comment:
Now that `GENERIC_END` positions are always reported correctly, and emitted accurately, this line was no longer covered. 

Example:
```java
        protected Pair<Integer, Pair<String, Pair<String, Object>>[]>[] values3a;
```

Difference between latest master and PR branch:
https://www.diffchecker.com/IuwJEwmB